src/callstack.cr

```diff
@@ -165,7 +172,27 @@ struct CallStack
         function = "???"
       end
 
-      "0x#{ip.address.to_s(16)}: #{function} at #{file_line_column}"
+      # We ignore these because they are part of `raise`'s internals,
+      # and we can't rely on a correct filename being available
+      # (for example if on Mac and without running `dsymutil`)
+      #
+      # We also ignore `main` because it's always at the same place
+      # and adds no info.
+      if function.starts_with?("*raise<") ||
+         function.starts_with?("*CallStack::") ||
+         function.starts_with?("*CallStack#")
+        next
+      end
+
+      # Crystal methods (their mangled name) start with `*`, so
+      # we remove that to have less clutter in the output.
+      function = function.lchop('*')
```

Comment:
All the unmangling of the function name could happen in the `elseif` above (line 168-170), when we failed to decode the function name from DWARF and got the mangled symbol instead.